src/__tests__/pinecone.test.ts

```diff
@@ -14,48 +16,38 @@ jest.mock('../utils', () => {
 });
 jest.mock('../data/fetch');
 jest.mock('../data/upsert');
+jest.mock('../data/hostUrlSingleton');
+
+jest.mock('../control', () => {
```

Comment:
I spent a lot of time trying to get the mocking for things working and it still feels like a mess. 😭